src/napari/__init__.py

```diff
@@ -14,6 +18,78 @@
 limit_numpy1x_threads_on_macos_arm()
 
 
+# TODO: probably move this elsewhere
+def register_napari_font():
+    import os  # TODO: WTF why do I need this here???
+
+    font_dir = Path(
+        resources.files(__package__).joinpath('resources', 'fonts')
+    )
+
+    if sys.platform == 'darwin':
+        from ctypes import POINTER, c_bool, c_char_p, c_uint32, c_void_p
+
+        from vispy.ext.cocoapy import cf, ct
+
+        # following vispy's implementation in vispy.ext.cocoapy
+        # and vispy.utils.fonts._quartz.py
+        cf.CFURLCreateFromFileSystemRepresentation.restype = c_void_p
+        cf.CFURLCreateFromFileSystemRepresentation.argtypes = [
+            c_void_p,
+            c_char_p,
+            c_uint32,
+            c_bool,
+        ]
+
+        ct.CTFontManagerRegisterFontsForURL.restype = c_bool
+        ct.CTFontManagerRegisterFontsForURL.argtypes = [
+            c_void_p,
+            c_uint32,
+            POINTER(c_void_p),
+        ]
+
+        # vispy/qt use quartz for font discovery on mac
+        for font_file in font_dir.glob('*.ttf'):
+            path = str(font_file).encode('utf-8')
+            url = cf.CFURLCreateFromFileSystemRepresentation(
+                None, path, len(path), False
+            )
+            _ok = ct.CTFontManagerRegisterFontsForURL(
+                url, 2, None
+            )  # 2 = scope user
```

Comment:
With scope 2 it fails to register the font

```suggestion
            ok = ct.CTFontManagerRegisterFontsForURL(
                url, 2, None
            )  # 1 = scope process
            if not ok:
                logging.error("Fail to register custom fonts")
```